api/envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -44,15 +46,29 @@ message DnsFilterConfig {
   // in a client context. This message will contain the timeouts, retry,
   // and forwarding configuration for Envoy to make DNS requests to other
   // resolvers
+  //
+  // [#next-free-field: 6]
   message ClientContextConfig {
     // Sets the maximum time we will wait for the upstream query to complete
     // We allow 5s for the upstream resolution to complete, so the minimum
     // value here is 1. Note that the total latency for a failed query is the
     // number of retries multiplied by the resolver_timeout.
     google.protobuf.Duration resolver_timeout = 1 [(validate.rules).duration = {gte {seconds: 1}}];
 
+    // This field was used for `dns_resolution_config` in Envoy 1.19.0 and
+    // 1.19.1.
+    // Control planes that need to set this field for Envoy 1.19.0 and
+    // 1.19.1 clients should fork the protobufs and change the field type
+    // to `DnsResolutionConfig`.
+    // Control planes that need to simultaneously support Envoy 1.18.x and
+    // Envoy 1.19.x should avoid Envoy 1.19.0 and 1.19.1.
+    //
+    // [#not-implemented-hide:]
+    repeated config.core.v3.Address upstream_resolvers = 2
+        [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
+
```

Comment:
Why not just reserve this number and field name?

Reply:
I'm assuming that there is a control plane that will consume these protobufs and have Envoy 1.18 and 1.19 clients. That's the case for Kuma (though IIUC we will end up consuming this change from Envoy master via the sync to go-control-plane).

Anyway, reserving the field fixes the envoy side in this branch, but the type is still needed so that a control plane can use this proto definition with 1.18.